perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -606,7 +662,10 @@ protected void removeDirectMember(PerunSession sess, Group group, Member member)
 		// check all relations with this group and call processRelationMembers to reflect changes of removing member from group
 		List<Integer> relations = groupsManagerImpl.getResultGroupsIds(sess, group.getId());
 		for (Integer groupId : relations) {
-			processRelationMembers(sess, groupsManagerImpl.getGroupById(sess, groupId), Collections.singletonList(member), group.getId(), false);
+			Group relationGroup =  groupsManagerImpl.getGroupById(sess, groupId);
+			//Create lock for affected group and member
+			PerunLocksUtils.lockGroupMembership(relationGroup, Arrays.asList(member));
```

Comment:
Both methods create immutable list of objects, but Collections.singletonList(Object) seems to be less complicated (it is just an immutable list of 1 Object) so it is probably more suited for this operation.